cloud/defaults.go

```diff
@@ -206,3 +208,14 @@ func GetDefaultUbuntuImage(k8sVersion string) (*infrav1.Image, error) {
 func UserAgent() string {
 	return fmt.Sprintf("cluster-api-provider-azure/%s", version.Get().String())
 }
+
+// SetAutoRestClientDefaults set authorizer and user agent for autorest client
+func SetAutoRestClientDefaults(c *autorest.Client, auth autorest.Authorizer) {
+	c.Authorizer = auth
+	AutoRestClientAppendUserAgent(c, UserAgent())
```

Comment:
`UserAgent` does not appear to be changeable, so it was not tested against `SetAutoRestClientDefaults`